exercises/practice/grade-school/GradeSchoolTest.php

```diff
@@ -41,26 +19,96 @@ protected function setUp(): void
+    /**
+     * uuid: 9337267f-7793-4b90-9b4a-8e3978408824
+     * @testdox Add a student
+     */
+    public function testAddAStudent(): void
+    {
+        $this->school->add('Aimee', 2);
+        $this->assertCount(1, $this->school->grade(2));
+    }
+
+    /**
+     * uuid: 6d0a30e4-1b4e-472e-8e20-c41702125667
+     * @testdox Student is added to the roster
+     */
+    public function testStudentIsAddedToTheRoster(): void
     {
-        $this->school->add('Claire', 2);
-        $this->assertContains('Claire', $this->school->grade(2));
+        $this->school->add('Aimee', 2);
+        $this->assertContains('Aimee', $this->school->grade(2));
     }
 
-    public function testAddStudentsInSameGrade(): void
+    /**
+     * uuid: 73c3ca75-0c16-40d7-82f5-ed8fe17a8e4a
+     * @testdox Adding multiple students in the same grade in the roster
+     */
+    public function testAddingMultipleStudentsInTheSameGradeInTheRoster(): void
     {
-        $this->school->add('Marc', 2);
-        $this->school->add('Virginie', 2);
-        $this->school->add('Claire', 2);
+        $this->school->add('Blair', 2);
+        $this->school->add('James', 2);
+        $this->school->add('Paul', 2);
 
         $students = $this->school->grade(2);
         $this->assertCount(3, $students);
         $this->assertEqualsCanonicalizing(
-            ['Claire', 'Marc', 'Virginie'],
+            ['Blair', 'James', 'Paul'],
             array_values($students)
         );
     }
 
+    /**
+     * uuid: 233be705-dd58-4968-889d-fb3c7954c9cc
+     * @testdox Multiple students in the same grade are added to the roster
+     */
+    public function testMultipleStudentsInTheSameGradeAreAddedToTheRoster(): void
+    {
+        $this->school->add('Blair', 2);
+        $this->school->add('James', 2);
+        $this->school->add('Paul', 2);
+
+        $students = $this->school->grade(2);
+        $this->assertCount(3, $students);
+        $this->assertEquals(
+            ['Blair', 'James', 'Paul'],
+            array_values($students)
+        );
+    }
+
+    /**
+     * uuid: 87c871c1-6bde-4413-9c44-73d59a259d83
+     * @testdox Cannot add student to same grade in the roster more than once
+     */
+    public function testCannotAddStudentToSameGradeMoreThanOnce(): void
+    {
+        $students = [
+            ["Blair", 2],
+            ["James", 2],
+            ["James", 2],
+            ["Paul", 2]
+        ];
+
+        $actual = [];
+        $roster = [];
+
+        $actual = array_map(function ($student) use (&$roster) {
+            [$name, $grade] = $student;
+
+            if (!isset($roster[$grade])) {
+                $roster[$grade] = [];
+            }
+
+            if (isset($roster[$grade][$name])) {
+                return false;
+            } else {
+                $roster[$grade][$name] = true;
+                return true;
+            }
+        }, $students);
+
+        $this->assertSame([true, true, false, true], $actual);
```

Comment:
This test does not contain `$this->school->add();`, so it does **not** test anything. I suggest:

```suggestion
        $actual = [];

        $actual[] = $this->school->add("Blair", 2);
        $actual[] = $this->school->add("James", 2);
        $actual[] = $this->school->add("James", 2);
        $actual[] = $this->school->add("Paul", 2);

        $this->assertSame([true, true, false, true], $actual);
```